### PR TITLE
feat: Fix voice over issues for conversation view Part 1 ACC-141

### PR DESCRIPTION
**Pull request by KaterinaWire**
<table>
<td>
  <a href="https://wearezeta.atlassian.net/browse/ACC-141" title="ACC-141" target="_blank">StoryACC-141</a>  Information, structure and relationships can be identified (1.3.1)
  </td></table>
  <br />
 

----
#### PR Submission Checklist for internal contributors

- The **PR Title**
  - [x] conforms to the style of semantic commits messages¹ supported in Wire's Github Workflow²
  - [x] contains a reference JIRA issue number like `SQPIT-764`
  - [ ] answers the question: _If merged, this PR will: ..._ ³

- The **PR Description**
  - [ ] is free of optional paragraphs and you have filled the relevant parts to the best of your ability
----

# What's new in this PR?

Fix voice over issues for conversation view

----
#### PR Post Submission Checklist for internal contributors (Optional)

 - [ ] Wire's Github Workflow has automatically linked the PR to a JIRA issue
----
#### PR Post Merge Checklist for internal contributors

 - [ ] If any soft of configuration variable was introduced by this PR, it has been added to the relevant documents and the CI jobs have been updated.
----
##### References
1. https://sparkbox.com/foundry/semantic_commit_messages
1. https://github.com/wireapp/.github#usage
1. E.g. `feat(conversation-list): Sort conversations by most emojis in the title #SQPIT-764`.

### CODE REVIEW

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift [hunk 1]
```diff
@@ -216,23 +216,25 @@ enum MessageAction: CaseIterable {
...
+            return MessageAction.EmojiButton.description
         case .showInConversation:
-            return "reveal in conversation"
+            return MessageAction.RevealButton.description
         case .delete:
-            return "delete"
+            return MessageAction.DeleteButton.description
         case .unlike:
-            return "unlike"
+            return MessageAction.LikeButton.description
```
Comment (MarcinR) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift:235]: is this correct? Looks like it should be MessageAction.UnlikeButton

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift [hunk 2]
```diff
@@ -216,23 +216,25 @@ enum MessageAction: CaseIterable {
...
         case .like:
-            return "like"
+            return MessageAction.UnlikeButton.description
```
Comment (MarcinR) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift:237]: as above

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift:235]: Yes, it's correct. It used to show the current state, but now it says about the action: "Like the picture"

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/MessageAction.swift:237]: Same as previous comment

**Review by MarcinR — APPROVED** [review]

File: Wire-iOS/Sources/UserInterface/Collections/CollectionAudioCell.swift [hunk 3]
```diff
@@ -52,6 +54,9 @@ final class CollectionAudioCell: CollectionCell {
             setup(restrictionView)
             restrictionView.configure()
         }
+
+        accessibilityLabel = ConversationSearch.SendBy.description(message.senderName) + ", \(message.serverTimestamp?.formattedDate ?? ""), " + ConversationSearch.AudioMessage.description
```
Comment (agisilaos) [inline Wire-iOS/Sources/UserInterface/Collections/CollectionAudioCell.swift:58]: Can we try and make this shorter? That's way too long.

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Ephemeral.swift [hunk 4]
```diff
@@ -98,6 +98,43 @@ extension ConversationInputBarViewController {
...
+
+private extension MessageDestructionTimeoutValue {
+
+    var accessibilityValue: String? {
+        typealias Conversation = L10n.Accessibility.Conversation
+
+        guard self != .none,
+              let timeoutValue = shortDisplayString else {
+                  return nil
+              }
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/ConversationInputBarViewController+Ephemeral.swift:121]:
```suggestion
        guard 
           self != .none,
           let timeoutValue = shortDisplayString
        else {
           return nil
        }
```

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift [hunk 5]
```diff
@@ -121,6 +122,22 @@ final class InputBarButtonsView: UIView {
...
         buttonRowTopInset.constant = CGFloat(rowIndex) * constants.buttonsBarHeight
         UIView.animate(easing: .easeInOutExpo, duration: animated ? 0.35 : 0, animations: layoutIfNeeded)
+        setupAccessibility()
+    }
+
+    private func setupAccessibility() {
+        if multilineLayout {
+            let firstRowButtons = [UIButton](buttons.prefix(customButtonCount))
+            let secondRowButtons = [UIButton](buttons.suffix(buttons.count - customButtonCount))
+            buttons.forEach { $0.isAccessibilityElement = currentRow == 0 ? firstRowButtons.contains($0) : secondRowButtons.contains($0) }
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift:132]: this is a little hard to read

Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift]: I tried to fix it. Could you check?

File: Wire-iOS/Sources/UserInterface/Collections/CollectionAudioCell.swift (hunk 3 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Collections/CollectionAudioCell.swift]: I moved it to separate lines

File: Wire-iOS/Sources/UserInterface/Collections/ConversationImagesViewController.swift [hunk 6]
```diff
@@ -109,6 +109,7 @@ final class ConversationImagesViewController: TintColorCorrectedViewController {
 
     override func viewWillAppear(_ animated: Bool) {
         super.viewWillAppear(animated)
+        navigationItem.rightBarButtonItem?.accessibilityLabel = L10n.Accessibility.PictureView.CloseButton.description
```
Comment (agisilaos) [inline Wire-iOS/Sources/UserInterface/Collections/ConversationImagesViewController.swift:112]: We can use typealias here.

File: Wire-iOS/Sources/UserInterface/Collections/CollectionsViewController.swift [hunk 7]
```diff
@@ -293,6 +293,7 @@ final class CollectionsViewController: UIViewController {
         navigationItem.titleView = titleViewWrapper
 
         let button = CollectionsView.closeButton()
+        button.accessibilityLabel = L10n.Accessibility.ConversationSearch.CloseButton.description
```
Comment (agisilaos) [inline Wire-iOS/Sources/UserInterface/Collections/CollectionsViewController.swift:296]: Same here.

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/CellComponents/SenderCellComponent.swift [hunk 8]
```diff
@@ -95,6 +95,11 @@ final class SenderCellComponent: UIView {
         avatar.size = .badge
         avatar.translatesAutoresizingMaskIntoConstraints = false
         avatar.addGestureRecognizer(UITapGestureRecognizer(target: self, action: #selector(tappedOnAvatar)))
+        avatar.accessibilityElementsHidden = false
```
Comment (agisilaos) [inline Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/CellComponents/SenderCellComponent.swift:98]: Same here.

**Review by agisilaos — COMMENTED** [review]
We can use typelias in some places. Let's try and do that. Can you also please take a look at a couple SwiftLint warnings regarding trailing?

File: Wire-iOS/Sources/UserInterface/Collections/ConversationImagesViewController.swift (hunk 6 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Collections/ConversationImagesViewController.swift:112]: Just for 1 string?

File: Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/CellComponents/SenderCellComponent.swift (hunk 8 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/UserInterface/Conversation/Content/Cells/CellComponents/SenderCellComponent.swift:98]: What do you mean here?

File: Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift [hunk 9]
```diff
@@ -121,6 +122,32 @@ final class InputBarButtonsView: UIView {
...
+            buttons.forEach {
+                var isAccessibilityElement = false
+                if currentRow == 0 {
+                    isAccessibilityElement = firstRowButtons.contains($0)
+                } else {
+                    isAccessibilityElement = secondRowButtons.contains($0)
+                }
+                $0.isAccessibilityElement = isAccessibilityElement
+
+            }
```
Comment (David-Henner) [inline Wire-iOS/Sources/UserInterface/Conversation/InputBar/InputBarButtonsView.swift:142]:
```suggestion
            buttons.forEach {
                $0.isAccessibilityElement = currentRow == 0 
                   ? firstRowButtons.contains($0)
                   : secondRowButtons.contains($0)
            }
```
just suggesting another way, feel free to choose what you prefer :)

**Review by David-Henner — APPROVED** [review]

**Pull request merged**